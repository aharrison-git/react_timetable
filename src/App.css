.app {
    display: grid;
    grid-template-columns: 150px 150px 150px 150px 150px 150px 150px 300px;
    grid-template-rows: 25px 25px 50px 50px 50px 50px 50px 50px 50px 50px 50px 50px 50px 50px 50px 50px 25px;
    justify-content: center;
    column-gap: 5px;
    row-gap: 5px;
}

.box {
    color: whitesmoke;
    text-align: center;
    font-weight: bold;
    width: 95%;
    height: 95%;
}

.daylabel {
    grid-row-start: 2;
    background-color: cadetblue;
}

.timelabel {
    grid-column-start: 1;
    background-color: darkslategrey;
}

.header {
    background-color: teal;
    grid-column-start: 1;
    grid-column-end: 8;
    grid-row-start: 1;
}

.footer {
    background-color: teal;
    grid-column-start: 1;
    grid-column-end: 8;
}


li {
    list-style-type: none;
    flex-grow: 1;
    
}

.monday {
    grid-column-start: 2;
    grid-column-end: 2;
}

.tuesday {
    grid-column-start: 3;
    grid-column-end: 4;
}

.session1 {
    grid-row-start: 3;
    grid-row-end: 3;
}
.session2 {
    grid-row-start: 4;
    grid-row-end: 4;
}
.session3 {
    grid-row-start: 5;
    grid-row-end: 5;
}
.session4 {
    grid-row-start: 6;
    grid-row-end: 6;
}
.session5 {
    grid-row-start: 7;
    grid-row-end: 7;
}
.session5 {
    grid-row-start: 7;
    grid-row-end: 7;
}

.session5 {
    grid-row-start: 7;
    grid-row-end: 7;
}

.session6 {
    grid-row-start: 8;
    grid-row-end: 8;
}

.session7 {
    grid-row-start: 9;
    grid-row-end: 9;
}

.session8 {
    grid-row-start: 10;
    grid-row-end: 10;
}

.session9 {
    grid-row-start: 11;
    grid-row-end: 11;
}

.day1 {
    grid-column-start: 2;
    grid-column-end: 2;
}

.day2 {
    grid-column-start: 3;
    grid-column-end: 3;
}

.day3 {
    grid-column-start: 4;
    grid-column-end: 4;
}

.day4 {
    grid-column-start: 5;
    grid-column-end: 5;
}

.day5 {
    grid-column-start: 6;
    grid-column-end: 6;
}

.day6 {
    grid-column-start: 7;
    grid-column-end: 7;
}

.day7 {
    grid-column-start: 8;
    grid-column-end: 8;
}

.ninjastars {
    background-color: midnightblue;
}

.samuraiwarriors {
    background-color: crimson;
}

.junioradults {
    background-color: black;
}

.filter {
    background-color: burlywood;
    grid-row-start: 2;
    grid-row-end: 7;
    grid-column-start: 8;
    grid-column-end: 11
}

.filter-checkbox {
    text-align: left;
}

.edit {
    background-color: burlywood;
    grid-row-start: 8;
    grid-row-end: 13;
    grid-column-start: 8;
    grid-column-end: 11; 
}
.editform {
    text-align: left;
}

.hidden {
    display: none;
}