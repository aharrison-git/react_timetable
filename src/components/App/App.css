.app {
    display: grid;
    grid-template-columns: 180px 180px 180px 180px 180px 180px 360px;
    grid-template-rows: 30px 30px 80px 80px 80px 80px 80px 80px 80px 30px;
    justify-content: center;
    column-gap: 2px;
    row-gap: 2px;
}

.box {
    color: whitesmoke;
    text-align: center;
    font-weight: bold;
    
}

.daylabel {
    grid-row-start: 2;
    background-color: cadetblue;
}

.timelabel {
    grid-column-start: 1;
    background-color: darkslategrey;
}

.header {
    background-color: teal;
    grid-column-start: 1;
    grid-column-end: 7;
    grid-row-start: 1;
}

.footer {
    background-color: teal;
    grid-column-start: 1;
    grid-column-end: 7;
}


li {
    list-style-type: none;
    flex-grow: 1;
    
}

.session1 {
    grid-row-start: 3;
    grid-row-end: 3;
}

.session2 {
    grid-row-start: 4;
    grid-row-end: 4;
}

.session3 {
    grid-row-start: 5;
    grid-row-end: 5;
}

.session4 {
    grid-row-start: 6;
    grid-row-end: 6;
}

.session5 {
    grid-row-start: 7;
    grid-row-end: 7;
}

.session6 {
    grid-row-start: 8;
    grid-row-end: 8;
}

.session7 {
    grid-row-start: 9;
    grid-row-end: 9;
}

.session8 {
    grid-row-start: 10;
    grid-row-end: 10;
}

.session9 {
    grid-row-start: 11;
    grid-row-end: 11;
}

.day1 {
    grid-column-start: 2;
    grid-column-end: 2;
}

.day2 {
    grid-column-start: 3;
    grid-column-end: 3;
}

.day3 {
    grid-column-start: 4;
    grid-column-end: 4;
}

.day4 {
    grid-column-start: 5;
    grid-column-end: 5;
}

.day5 {
    grid-column-start: 6;
    grid-column-end: 6;
}

.english {
    background-color: midnightblue;
}

.maths {
    background-color: crimson;
}

.physics {
    background-color: black;
}

.chemistry {
    background-color: green;
}

.music {
    background-color: goldenrod;
}


.edit {
    background-color: teal;
    grid-row-start: 7;
    grid-row-end: 11;
    grid-column-start: 7;
    grid-column-end: 9; 
}



.login {
    background-color: teal;
    grid-row-start: 1;
    grid-row-end: 5;
    grid-column-start: 7;
    grid-column-end: 9;  
}

.login-form {
    display: grid;
    grid-template-columns: 100px 180px;
    grid-template-rows: repeat(4, 30px);
}

.login-row1 {
    grid-row-start: 1;
    grid-row-end: 1;
}

.login-row2 {
    grid-row-start: 2;
    grid-row-end: 2;
}

.login-row3 {
    grid-row-start: 3;
    grid-row-end: 3;
}

.login-column1 {
    grid-column-start: 1;
    grid-column-end: 1;
}

.login-column2 {
    grid-column-start: 2;
    grid-column-end: 2;
}

.login button {
    width: 50%;
}
.errormessage {
    color: #eb262c;
    font-family: Cambria, Cochin, Georgia, Times, 'Times New Roman', serif;
}

.filter {
    background-color: teal;
    grid-row-start: 5;
    grid-row-end: 7;
    grid-column-start: 7;
    grid-column-end: 9;
    
}


.hidden {
    display: none;
}