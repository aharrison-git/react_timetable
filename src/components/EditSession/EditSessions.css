.editform {
    text-align: left;
    display: grid;
    grid-template-columns: 120px 200px;
    grid-template-rows: repeat(4, 30px) 40px;
}

.cell {
    margin-left: 10px;
}

.row1 {
    grid-row-start: 1;
    grid-row-end: 1;
}

.row2 {
    grid-row-start: 2;
    grid-row-end: 2;
}

.row3 {
    grid-row-start: 3;
    grid-row-end: 3;
}

.row4 {
    grid-row-start: 4;
    grid-row-end: 4;
}

.row5 {
    grid-row-start: 5;
    grid-row-end: 5;
    width: 50%;
}

.column1 {
    grid-column-start: 1;
    grid-column-end: 1;
}

.column2 {
    grid-column-start: 2;
    grid-column-end: 2;
}

.success {
    color:  lightgreen;
    font-weight: normal;
}

.error {
    color: red;
    background-color: lightpink;
    font-weight: normal;
}