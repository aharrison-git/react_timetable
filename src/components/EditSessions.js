import React from 'react'
import groupData from '../data'
import GroupTypes from '../GroupTypes'

class EditSessions extends React.Component {
    constructor(props) {
        super(props)
        this.state = {class: GroupTypes[0], day:"Monday", time: "9.00", action: "Add"}
        this.handleChangeClass = this.handleChangeClass.bind(this)
        this.handleChangeDay = this.handleChangeDay.bind(this)
        this.handleChangeTime = this.handleChangeTime.bind(this)
        this.handleChangeAction = this.handleChangeAction.bind(this)
        this.handleSubmit = this.handleSubmit.bind(this)
    }

    handleChangeClass(event) {
        this.setState({class: event.target.value})
        console.log("target: " + event.target.value)
    }

    handleChangeDay(event) {
        this.setState({day: event.target.value})
        console.log("target: " + event.target.value)
    }

    handleChangeTime(event) {
        this.setState({time: event.target.value})
        console.log("target: " + event.target.value)
    }

    handleChangeAction(event) {
        this.setState({action: event.target.value})
        console.log("target: " + event.target.value)
    }

    handleSubmit(event) {
        alert("class: " + this.state.class + ", day: " + this.state.day + ", time: " + this.state.time + ", action: " + this.state.action)
        event.preventDefault()
        this.props.parentCallback(this.state)
    }

    render() {
        const classOptions = GroupTypes.map((name) => 
            <option value={name}>{name}</option>)
        const days = ["Monday", "Tuesday", "Wednesday"].map((day) =>
            <option value={day}>{day}</option>)
        const times = ["9.00", "9.45", "10.30", "11.15"].map((time) =>
            <option value={time}>{time}</option>)
        const actions = ["Add", "Delete"].map((action) =>
            <option value={action}>{action}</option>)

        return(
            <div>
                <p>Edit Sessions</p>
                
                <form class="editform" onSubmit={this.handleSubmit}>
                    <label>
                        Select Class:
                        <select value={this.state.class} onChange={this.handleChangeClass}>            
                            {classOptions}
                        </select>
                    </label>
                    <br/>
                    <label>
                        Select Day:
                        <select value={this.state.day} onChange={this.handleChangeDay}>            
                            {days}
                        </select>
                    </label>
                    <br/>
                    <label>
                        Select Time:
                        <select value={this.state.time} onChange={this.handleChangeTime}>            
                            {times}
                        </select>
                    </label>
                    <br/>
                    <label>
                        Select Action:
                        <select value={this.state.action} onChange={this.handleChangeAction}>            
                            {actions}
                        </select>
                    </label>
                    <br/>
                    
                    <input type="submit" value="Apply" />
                </form>
            </div>
        )
    }
}
//<option value="grapefruit">Grapefruit</option>
//<option value="lime">Lime</option>
//<option value="coconut">Coconut</option>
//<option value="mango">Mango</option>
export default EditSessions
