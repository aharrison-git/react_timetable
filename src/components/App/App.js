import React from 'react';
import './App.css';
import Filter from '../Filter/Filter'
import AllSessions from '../AllSessions/AllSessions'
import EditSession from '../EditSession/EditSessions'
import groupTypes from '../../static/GroupTypes'
import daysOpen from '../../static/days'
import sessionTimes from '../../static/sessionTimes'

class App extends React.Component {
  constructor() {
    super()
    const groups = groupTypes.map((group) => ({group: true}) )
    this.state = {filter: {
      maths: true,
      english: true,
      physics: true,
      chemistry: true,
      music: true
      }, 
      sessions: {},
      isLoggedIn: false,
      username: '',
      password: '',
      showLoginFailMessage: false,
    }
    this.callbackFunctionFilter = this.callbackFunctionFilter.bind(this)
    this.callbackFunctionSessions = this.callbackFunctionSessions.bind(this)
    this.handleLoginSubmit = this.handleLoginSubmit.bind(this)
    this.handleLoginInput = this.handleLoginInput.bind(this)
  }

  callbackFunctionFilter(filters) {
    this.setState({filter: filters})
  }

  callbackFunctionSessions(sessionData) {
    this.setState({sessions: sessionData})
  }

  handleLoginInput(event) {
    let name = event.target.name
    let value = event.target.value
    this.setState({[name]: value})
    }

  handleLoginSubmit(event) {
    if (!this.state.isLoggedIn) {
      if (this.state.username == 'admin' && this.state.password == 'password') {
        this.setState({isLoggedIn: true})
      }
      else {
        this.setState({isLoggedIn: false})
        this.setState({showLoginFailMessage: true})
      }
    }
    else {
      this.setState({isLoggedIn: false})
      this.setState({username: ''})
      this.setState({showLoginFailMessage: false})
    }
    this.setState({password: ''})
    event.preventDefault()
  }

  render() {
    let editDiv
    let welcomeMessage = ''
    let loginFailDiv
    if (this.state.isLoggedIn) {
      editDiv = <div className="box edit">
                  <EditSession parentCallback={this.callbackFunctionSessions}/>
                </div>
      welcomeMessage = 'Welcome, ' + this.state.username
    }
    else { 
      editDiv = <div></div>
      welcomeMessage = ''
    }
    
    if (this.state.showLoginFailMessage && !this.state.isLoggedIn) {
      loginFailDiv = <p className="errormessage">login failed!</p>
    }
    else {
      loginFailDiv = <div></div>
    }

    const dayElements = daysOpen.map((day, index) => 
      <div key={index} className="box daylabel">{day}</div>  
    )
    const sessionElements = sessionTimes.map((session, index) => <div key={index} className="box timelabel">{session.start} - {session.end}</div>)
    
    return (
      <div className="app">
        <div className="box header">Timetable</div>
        <div className="box daylabel">Session Times</div>  
        {dayElements}
        {sessionElements}      

        <AllSessions filter={this.state.filter} />
        
        <div className="box login">
          <p>Staff Login</p>
          <div>
            <form className="login-form" onSubmit={this.handleLoginSubmit}>
              <label className="login-row1 login-column1" for="username">Username</label>
              <label className="login-row2 login-column1" for="password">Password</label>
              <input className="login-row1 login-column2" type="text" value={this.state.username} name="username" onChange={this.handleLoginInput}></input>
              <input className="login-row2 login-column2" type="password" value={this.state.password} name="password" onChange={this.handleLoginInput}></input>
              <button className="btn login-row3 login-column2">{this.state.isLoggedIn? 'Logout' : 'Login'}</button>
            </form>
            {loginFailDiv}
            <div name="welcomeMessage">{welcomeMessage}</div> 
          </div>
        </div>

        <div className="box filter">
          <Filter parentCallback={this.callbackFunctionFilter}/>
          </div>
        
        {editDiv}

        <div className="box footer"></div>

      </div>
    )
  }
}
export default App;
