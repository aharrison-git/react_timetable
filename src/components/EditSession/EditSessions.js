import React from 'react'
import './EditSessions.css'
import groupData from '../../static/data'
import groupTypes from '../../static/GroupTypes'
import sessionTimes from '../../static/sessionTimes'
import daysOpen from '../../static/days'

class EditSessions extends React.Component {
    constructor(props) {
        super(props)
        this.state = {data: groupData, class: groupTypes[0], day:"Monday", time: "9.00", action: "Add"}
        this.handleChangeSelect = this.handleChangeSelect.bind(this)
        this.handleSubmit = this.handleSubmit.bind(this)
        this.daysOfTheWeekObject = daysOpen.map((element, index) => ({key: element, value: index}) )
        this.sessionTimes = sessionTimes.map((session) => session.start + " - " + session.end)
        this.sessionTimesObject = this.sessionTimes.map((element, index) => ({key: element, value: index}))
        this.getCode = this.getCode.bind(this)
        this.getSessionCodeString = this.getSessionCodeString.bind(this)
        this.updateData = this.updateData.bind(this)
    }

    getCode(value, objectArray) {
        for (var i=0 ; i<objectArray.length ; i++) {
            if (objectArray[i].key === value) {
                return objectArray[i].value
            }
        } 
        return null
    }


    getSessionCodeString(time, day) {
        const dayCode = this.getCode(this.state.day, this.daysOfTheWeekObject)
        const timeCode = this.getCode(this.state.time, this.sessionTimesObject)
        var sessionCode = (timeCode + 1).toString() + (dayCode + 1).toString()
        return sessionCode
    }


    updateData(name, session, action) {
        switch (action) {
            case "Add":
                var exists = 0
                for (var i=0; i<this.state.data.length; i++) {
                    if (this.state.data[i].groupName == name && this.state.data[i].session == session) {
                        exists++    
                    }
                }
                if (exists===0) {
                    var dataTemp = this.state.data
                    const newObj = {session: parseInt(session), groupName: name}
                    dataTemp.push(newObj)
                }
                break

            case "Delete":
                exists = 0
                for (i=0; i<this.state.data.length; i++) {
                    console.log(this.state.data[i].groupName + " " + this.state.data[i].session)
                    if (this.state.data[i].groupName == name && this.state.data[i].session == session) {
                        dataTemp = this.state.data.splice(i, 1)
                        break
                    }
                }
                break

            default:
                break
        }
    }

    handleChangeSelect(event) {
        this.setState({[event.target.name]: event.target.value})
    }

    handleSubmit(event) {
        const sessionCodeString = this.getSessionCodeString(this.state.time, this.state.day)
        this.updateData(this.state.class, sessionCodeString, this.state.action)
        event.preventDefault()
        this.props.parentCallback(this.state)
       }


    render() {
        const classOptions = groupTypes.map((name, index) => 
            <option key={index} value={name}>{name}</option>)
        const days = daysOpen.map((day, index) =>
            <option key={index} value={day}>{day}</option>)
        const times = this.sessionTimes.map((time, index) =>
            <option key={index} value={time}>{time}</option>)
        const actions = ["Add", "Delete"].map((action, index) =>
            <option key={index} value={action}>{action}</option>)

        return(
            <div>
                <p>Edit Sessions</p>
                
                <form className="editform" onSubmit={this.handleSubmit}>
                    <label className="cell row1 column1" for="class">Select Class</label>
                    <label className="cell row2 column1" for="day">Select Day</label>
                    <label className="cell row3 column1" for="class">Select Time</label>
                    <label className="cell row4 column1" for="class">Select Action</label>
                    <select className="cell row1 column2" value={this.state.class} name="class" onChange={this.handleChangeSelect}>{classOptions}</select>
                    <select className="cell row2 column2" value={this.state.day} name="day" onChange={this.handleChangeSelect}>{days}</select>
                    <select className="cell row3 column2" value={this.state.time} name="time" onChange={this.handleChangeSelect}>{times}</select>
                    <select className="cell row4 column2" value={this.state.action} name="action" onChange={this.handleChangeSelect}>{actions}</select>
                    <input className="cell column2 row5" type="submit" value="Apply" />
                </form>
            </div>
        )
    }
}
export default EditSessions
