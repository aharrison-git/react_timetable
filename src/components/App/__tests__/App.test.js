import React from 'react'
import App from '../App'
import { create } from 'react-test-renderer'
import { shallow, mount } from 'enzyme'
import EditSessions from '../../EditSession/EditSessions'




describe("Snapshot testing for App component", () => {
    test("snapshot test for App component", () => {
        let tree = create(<App />)
        expect(tree.toJSON()).toMatchSnapshot()
    })
})

describe("Test Login renders", () => {
    const wrapper = shallow(<App />)

    test("login form renders", () => {
        //const wrapper = shallow(<App />)
        expect(wrapper.find('form.login-form').exists()).toBe(true)
    })
    test("username field renders", () => {
        expect(wrapper.find('input[name="username"]').exists()).toBe(true)
    })
    test("password field renders", () => {
        expect(wrapper.find('input[name="password"]').exists()).toBe(true)
    })
})

describe("Test Login functionality", () => {

    test("username text updates state", () => {
        const wrapper = shallow(<App />)
        const username = 'admin'
        wrapper.find('input[name="username"]').simulate('change', {target: {name: 'username', value: username}})
        expect(wrapper.state('username')).toEqual(username)
    })
    test("password text updates state", () => {
        const wrapper = shallow(<App />)
        const password = 'password'
        wrapper.find('input[name="password"]').simulate('change', {target: {name: 'password', value: password}})
        expect(wrapper.state('password')).toEqual(password)
    })
    test("button text is 'Login' when component first renders", () => {
        const wrapper = shallow(<App />)
        const button = wrapper.find('button')
        expect(button.text()).toEqual('Login');
    })

    test("state reflects user is not logged in when component first renders", () => {
        const wrapper = shallow(<App />)
        expect(wrapper.state('isLoggedIn')).toEqual(false)
        expect(wrapper.state('username')).toEqual('')
        expect(wrapper.state('showLoginFailMessage')).toBe(false)
    })
    
    test("state changes when user is logged", () => {
        const user = 'admin'
        const pwd = 'password'
        const wrapper = shallow(<App />)
        login(user, pwd, wrapper)
        expect(wrapper.state('isLoggedIn')).toEqual(true)
        expect(wrapper.state('password')).toEqual('')
    })

    test("welcome message displayed when user is logged in", () => {
        const user = 'admin'
        const pwd = 'password'
        const wrapper = shallow(<App />)
        login(user, pwd, wrapper)
        const message = wrapper.find('div[name="welcomeMessage"]')
        expect(message.text()).toEqual("Welcome, " + user)
    })

    test("user is not logged in when credentails are incorrect", () => {
        const user = 'admin'
        const pwd = 'badpassword'
        const wrapper = shallow(<App />)
        login(user, pwd, wrapper)
        expect(wrapper.state('isLoggedIn')).toBe(false)
    })

    test("error message is displayed when credentials are incorrect", () => {
        const user = 'admin'
        const pwd = 'badpassword'
        const wrapper = shallow(<App />)
        login(user, pwd, wrapper)
        const errorMessage = wrapper.find('.errormessage').text()
        expect(wrapper.state('showLoginFailMessage')).toBe(true)
        expect(errorMessage).toEqual('login failed!')
    })
})

describe("Logout functionality", () => {
    beforeEach(() => {
        const user = 'admin'
        const pwd = 'password'
        const wrapper = shallow(<App />)
        login(user, pwd, wrapper)
    })

    test("logged in user can logout successfully", () => { 
        const wrapper = shallow(<App />)
        const form = wrapper.find('.login-form')
        form.simulate('submit', {preventDefault: () => {} })
        expect(wrapper.state('isLoggedIn')).toBe(false)
    })

    test("state changes when user logs out", () => { 
        const wrapper = shallow(<App />)
        const form = wrapper.find('.login-form')
        form.simulate('submit', {preventDefault: () => {} })
        expect(wrapper.state('username')).toBe('')
    })

    test("button text changes to 'login' when user logs out", () => { 
        const wrapper = shallow(<App />)
        const form = wrapper.find('.login-form')
        form.simulate('submit', {preventDefault: () => {} })
        const button = wrapper.find('button')
        expect(button.text()).toBe('Login')
    })

    test("username field clears when user logs out", () => { 
        const wrapper = shallow(<App />)
        const form = wrapper.find('.login-form')
        form.simulate('submit', {preventDefault: () => {} })
        const username = wrapper.find('input[name="username"]')
        expect(username.text()).toEqual('')
    })
})


describe("Edit Session component", () => {
    test("Edit Session component is not displayed App component renders for first time", () => {
        const user = 'admin'
        const pwd = 'password'
        const wrapper = shallow(<App />)
        const component = wrapper.find('.edit')
        expect(wrapper.find(EditSessions)).toHaveLength(0)
    })

    test("Edit Session component is displayed when user is logged in", () => {
        const user = 'admin'
        const pwd = 'password'
        const wrapper = shallow(<App />)
        login(user, pwd, wrapper)
        const component = wrapper.find('.edit')
        expect(wrapper.state('isLoggedIn')).toBe(true)
        expect(wrapper.find(EditSessions)).toHaveLength(1)
    })

    test("Edit Session component is not displayed after user logs out", () => {
        const user = 'admin'
        const pwd = 'password'
        const wrapper = shallow(<App />)
        login(user, pwd, wrapper)
        
        // logout
        const form = wrapper.find('.login-form')
        form.simulate('submit', {preventDefault: () => {} })
        
        const component = wrapper.find('.edit')
        expect(wrapper.find(EditSessions)).toHaveLength(0)
    })
})

function login(username, password, wrapper) {
    const fakeEvent = { preventDefault: () => {} }
    wrapper.find('input[name="username"]').simulate('change', {target: {name: 'username', value: username}})
    wrapper.find('input[name="password"]').simulate('change', {target: {name: 'password', value: password}})
    const form = wrapper.find('.login-form')
    form.simulate('submit', fakeEvent)
}







// * App.test.js * original
// import React from 'react';
// import { render } from '@testing-library/react';
// import App from './App';

// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
