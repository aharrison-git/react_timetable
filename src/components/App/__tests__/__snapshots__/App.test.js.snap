// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing for App component snapshot test for App component 1`] = `
<div
  className="app"
>
  <div
    className="box header"
  >
    Timetable
  </div>
  <div
    className="box daylabel"
  >
    Session Times
  </div>
  <div
    className="box daylabel"
  >
    Monday
  </div>
  <div
    className="box daylabel"
  >
    Tuesday
  </div>
  <div
    className="box daylabel"
  >
    Wednesday
  </div>
  <div
    className="box daylabel"
  >
    Thursday
  </div>
  <div
    className="box daylabel"
  >
    Friday
  </div>
  <div
    className="box timelabel"
  >
    9.00
     - 
    9.45
  </div>
  <div
    className="box timelabel"
  >
    9.45
     - 
    10.30
  </div>
  <div
    className="box timelabel"
  >
    10.30
     - 
    11.15
  </div>
  <div
    className="box timelabel"
  >
    11.15
     - 
    12.00
  </div>
  <div
    className="box timelabel"
  >
    12.00
     - 
    12.45
  </div>
  <div
    className="box timelabel"
  >
    1.30
     - 
    2.15
  </div>
  <div
    className="box timelabel"
  >
    2.15
     - 
    3.00
  </div>
  <div
    className="box session1 day1"
  >
    <li
      className=" english "
      style={
        Object {
          "height": "calc(100% / 3)",
        }
      }
    >
      English
    </li>
    <li
      className=" physics "
      style={
        Object {
          "height": "calc(100% / 3)",
        }
      }
    >
      Physics
    </li>
    <li
      className=" maths "
      style={
        Object {
          "height": "calc(100% / 3)",
        }
      }
    >
      Maths
    </li>
  </div>
  <div
    className="box session2 day1"
  >
    <li
      className=" physics "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      Physics
    </li>
  </div>
  <div
    className="box session3 day1"
  >
    <li
      className=" maths "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      Maths
    </li>
  </div>
  <div
    className="box session5 day1"
  >
    <li
      className=" physics "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      Physics
    </li>
  </div>
  <div
    className="box session6 day1"
  >
    <li
      className=" chemistry "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      Chemistry
    </li>
  </div>
  <div
    className="box session7 day1"
  >
    <li
      className=" maths "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      Maths
    </li>
  </div>
  <div
    className="box session1 day2"
  >
    <li
      className=" chemistry "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      Chemistry
    </li>
  </div>
  <div
    className="box session2 day2"
  >
    <li
      className=" music "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      Music
    </li>
  </div>
  <div
    className="box session3 day2"
  >
    <li
      className=" maths "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      Maths
    </li>
  </div>
  <div
    className="box session4 day2"
  >
    <li
      className=" english "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      English
    </li>
  </div>
  <div
    className="box session6 day2"
  >
    <li
      className=" english "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      English
    </li>
  </div>
  <div
    className="box session7 day2"
  >
    <li
      className=" maths "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      Maths
    </li>
  </div>
  <div
    className="box session1 day3"
  >
    <li
      className=" maths "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      Maths
    </li>
  </div>
  <div
    className="box session2 day3"
  >
    <li
      className=" physics "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      Physics
    </li>
  </div>
  <div
    className="box session3 day3"
  >
    <li
      className=" english "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      English
    </li>
  </div>
  <div
    className="box session4 day3"
  >
    <li
      className=" music "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      Music
    </li>
  </div>
  <div
    className="box session6 day3"
  >
    <li
      className=" physics "
      style={
        Object {
          "height": "calc(100% / 2)",
        }
      }
    >
      Physics
    </li>
    <li
      className=" chemistry "
      style={
        Object {
          "height": "calc(100% / 2)",
        }
      }
    >
      Chemistry
    </li>
  </div>
  <div
    className="box session7 day3"
  >
    <li
      className=" physics "
      style={
        Object {
          "height": "calc(100% / 2)",
        }
      }
    >
      Physics
    </li>
    <li
      className=" chemistry "
      style={
        Object {
          "height": "calc(100% / 2)",
        }
      }
    >
      Chemistry
    </li>
  </div>
  <div
    className="box session1 day4"
  >
    <li
      className=" english "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      English
    </li>
  </div>
  <div
    className="box session2 day4"
  >
    <li
      className=" chemistry "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      Chemistry
    </li>
  </div>
  <div
    className="box session4 day4"
  >
    <li
      className=" english "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      English
    </li>
  </div>
  <div
    className="box session5 day4"
  >
    <li
      className=" music "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      Music
    </li>
  </div>
  <div
    className="box session6 day4"
  >
    <li
      className=" physics "
      style={
        Object {
          "height": "calc(100% / 2)",
        }
      }
    >
      Physics
    </li>
    <li
      className=" chemistry "
      style={
        Object {
          "height": "calc(100% / 2)",
        }
      }
    >
      Chemistry
    </li>
  </div>
  <div
    className="box session7 day4"
  >
    <li
      className=" physics "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      Physics
    </li>
  </div>
  <div
    className="box session1 day5"
  >
    <li
      className=" maths "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      Maths
    </li>
  </div>
  <div
    className="box session2 day5"
  >
    <li
      className=" physics "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      Physics
    </li>
  </div>
  <div
    className="box session4 day5"
  >
    <li
      className=" english "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      English
    </li>
  </div>
  <div
    className="box session5 day5"
  >
    <li
      className=" music "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      Music
    </li>
  </div>
  <div
    className="box session6 day5"
  >
    <li
      className=" music "
      style={
        Object {
          "height": "calc(100% / 2)",
        }
      }
    >
      Music
    </li>
    <li
      className=" maths "
      style={
        Object {
          "height": "calc(100% / 2)",
        }
      }
    >
      Maths
    </li>
  </div>
  <div
    className="box session7 day5"
  >
    <li
      className=" english "
      style={
        Object {
          "height": "calc(100% / 1)",
        }
      }
    >
      English
    </li>
  </div>
  <div
    className="box login"
  >
    <p>
      Staff Login
    </p>
    <div>
      <form
        className="login-form"
        onSubmit={[Function]}
      >
        <label
          className="login-row1 login-column1"
          for="username"
        >
          Username
        </label>
        <label
          className="login-row2 login-column1"
          for="password"
        >
          Password
        </label>
        <input
          className="login-row1 login-column2"
          name="username"
          onChange={[Function]}
          type="text"
          value=""
        />
        <input
          className="login-row2 login-column2"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
        <button
          className="btn login-row3 login-column2"
        >
          Login
        </button>
      </form>
      <div />
      <div
        name="welcomeMessage"
      >
        
      </div>
    </div>
  </div>
  <div
    className="box filter"
  >
    <div
      className="filter box"
    >
      <p>
        Filter
      </p>
      <div>
        <form
          className="filter-checkbox"
          onSubmit={[Function]}
        >
          <div
            className="checkbox1"
          >
            <input
              checked={true}
              name="maths"
              onChange={[Function]}
              type="checkbox"
            />
            <label>
              Maths
            </label>
          </div>
          <div
            className="checkbox2"
          >
            <input
              checked={true}
              name="english"
              onChange={[Function]}
              type="checkbox"
            />
            <label>
              English
            </label>
          </div>
          <div
            className="checkbox3"
          >
            <input
              checked={true}
              name="physics"
              onChange={[Function]}
              type="checkbox"
            />
            <label>
              Physics
            </label>
          </div>
          <div
            className="checkbox4"
          >
            <input
              checked={true}
              name="chemistry"
              onChange={[Function]}
              type="checkbox"
            />
            <label>
              Chemistry
            </label>
          </div>
          <div
            className="checkbox5"
          >
            <input
              checked={true}
              name="music"
              onChange={[Function]}
              type="checkbox"
            />
            <label>
              Music
            </label>
          </div>
          <button
            className="button"
          >
            Apply
          </button>
        </form>
      </div>
    </div>
  </div>
  <div />
  <div
    className="box footer"
  />
</div>
`;
